x-pack/legacy/plugins/siem/public/components/resize_handle/index.tsx

```diff
@@ -52,99 +48,76 @@ export const removeGlobalResizeCursorStyleFromBody = () => {
   document.body.classList.remove(globalResizeCursorClassName);
 };
 
-export class Resizeable extends React.PureComponent<Props, State> {
-  private drag$: Observable<MouseEvent> | null;
-  private dragEventTargets: Array<{ htmlElement: HTMLElement; prevCursor: string }>;
-  private dragSubscription: Subscription | null;
-  private prevX: number = 0;
-  private ref: React.RefObject<HTMLElement>;
-  private upSubscription: Subscription | null;
-
-  constructor(props: Props) {
-    super(props);
-
-    // NOTE: the ref and observable below are NOT stored in component `State`
-    this.ref = React.createRef<HTMLElement>();
-    this.drag$ = null;
-    this.dragSubscription = null;
-    this.upSubscription = null;
-    this.dragEventTargets = [];
-
-    this.state = {
-      isResizing: false,
-    };
-  }
-
-  public componentDidMount() {
-    const { id, onResize } = this.props;
-
+export const Resizeable = React.memo<Props>(({ id, onResize, handle, height, render }) => {
+  const drag$ = useRef<Observable<MouseEvent> | null>(null);
+  const dragEventTargets = useRef<Array<{ htmlElement: HTMLElement; prevCursor: string }>>([]);
+  const dragSubscription = useRef<Subscription | null>(null);
+  const prevX = useRef<number>(0);
+  const ref = useRef<React.RefObject<HTMLElement>>(React.createRef<HTMLElement>());
+  const upSubscription = useRef<Subscription | null>(null);
+  const [isResizing, setIsResizing] = useState<boolean>(false);
+
+  const calculateDelta = (e: MouseEvent) => {
```

Comment:
CalculateDelta is not working anymore the same way, it was like that before
```
  const calculateDelta = (e: MouseEvent) => {
    const deltaX = calculateDeltaX({ prevX: prevX.current, screenX: e.screenX });
    prevX.current = e.screenX;
    return deltaX;
  };
```
and you can see we were saving the previous screenX value in the prevX after calculating the deltaX. We still need to do the same thing.

Reply:
i see what you mean, ill change it back